src/object.cr

```diff
@@ -58,13 +58,19 @@ class Object
     nil
   end
 
-  # Generates an `Int` hash value for this object.
+  # Generates an `Int` hash value for this object (usually `UInt32`)
   #
   # This method must have the property that `a == b` implies `a.hash == b.hash`.
   #
   # The hash value is used along with `==` by the `Hash` class to determine if two objects
   # reference the same hash key.
   abstract def hash
+  def hash
+    StdHasher.hashit self
+  end
+
+  # Protocol method for safe hashing
+  abstract def hashme(hasher)
```

Comment:
This should be renamed to just `hash(hasher)` and let method overloading do the work. The precedent is in how `to_s` and `to_json` work.